plugins/k8saudit/pkg/k8saudit/source.go

```diff
@@ -211,7 +211,10 @@ func (k *Plugin) openEventSource(ctx context.Context, eventChan <-chan []byte, e
 				}
 			case <-ctx.Done():
 				return
-			case err := <-errorChan:
+			case err, ok := <-errorChan:
+				if !ok {
+					return
+				}
```

Comment:
Although this works because those channels are private, I think just one channel should be responsible for shutting down the goroutine. Anyway, it's ok for now. I have left this comment just as a reminder for future improvements.